processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -873,127 +941,67 @@ private static void makeDimColumn(
-                                {
-                                  return input.size();
-                                }
-                              },
-                              dictionarySize
-                          );
-                        }
+              Iterables.transform(
+                  vals,
+                  new Function<List<Integer>, VSizeIndexedInts>()
+                  {
+                    @Override
+                    public VSizeIndexedInts apply(final List<Integer> input)
+                    {
+                      if (input == null) {
+                        return VSizeIndexedInts.fromList(ImmutableList.<Integer>of(0), dictionarySize);
+                      } else {
+                        return VSizeIndexedInts.fromList(new NullsAtZeroConvertingIntList(input, false), dictionarySize);
                       }
-                  )
+                    }
+                  }
+              )
           );
         } else {
           final int dictionarySize = dictionary.size();
           singleValCol = null;
           multiValCol = VSizeIndexed.fromIterable(
-              FunctionalIterable
-                  .create(vals)
-                  .transform(
-                      new Function<List<Integer>, VSizeIndexedInts>()
-                      {
-                        @Override
-                        public VSizeIndexedInts apply(List<Integer> input)
-                        {
-                          if (input == null) {
-                            return VSizeIndexedInts.fromList(
-                                new AbstractList<Integer>()
-                                {
-                                  @Override
-                                  public Integer get(int index)
-                                  {
-                                    return 0;
-                                  }
-
-                                  @Override
-                                  public int size()
-                                  {
-                                    return 1;
-                                  }
-                                }, dictionarySize
-                            );
-                          }
-                          return VSizeIndexedInts.fromList(
-                              input,
-                              dictionarySize
-                          );
-                        }
+              Iterables.transform(
+                  vals,
+                  new Function<List<Integer>, VSizeIndexedInts>()
+                  {
+                    @Override
+                    public VSizeIndexedInts apply(List<Integer> input)
+                    {
+                      if (input == null) {
+                        return VSizeIndexedInts.fromList(ImmutableList.<Integer>of(0), dictionarySize);
+                      } else {
+                        return VSizeIndexedInts.fromList(input, dictionarySize);
```

Comment:
the original code also uses `VSizeIndexedInts.fromList(input, dictionarySize);` This PR is not trying to change the behavior in that case.